### PR TITLE
Fix time varying J and tcl injection in `Xf_ThermSpinXferEvolver`

### PR DESCRIPTION
**Pull request by swapneelap**

### CODE REVIEW

**Comment by lang-m** [comment]
Is any of the tests actually running a simulation with time-varying `Xf_ThermSpinXferEvolver`? If not, we should add a test to `micromagnetictests`.

File: oommfc/scripts/evolver.py [hunk 1]
```diff
@@ -69,7 +69,7 @@ def evolver_script(evolver, **kwargs):
         mif += "  return $factor\n"
         mif += "}\n\n"
 
-        if isinstance(evolver, (oc.SpinXferEvolver)):
+        if isinstance(evolver, (oc.SpinXferEvolver, oc.Xf_ThermSpinXferEvolver)):
             # oc.Xf_TherrmSpinXferEvolver)):
```
Comment (lang-m) [inline oommfc/scripts/evolver.py:73]:
```suggestion

```

File: oommfc/scripts/evolver.py [hunk 2]
```diff
@@ -79,7 +79,7 @@ def evolver_script(evolver, **kwargs):
     if hasattr(evolver, "tcl_strings") and isinstance(evolver.tcl_strings, dict):
         print(evolver.tcl_strings)
         mif += evolver.tcl_strings["script"]
-        if isinstance(evolver, (oc.SpinXferEvolver)):
+        if isinstance(evolver, (oc.SpinXferEvolver, oc.Xf_ThermSpinXferEvolver)):
             # oc.Xf_ThermSpinXferEvolver)):
```
Comment (lang-m) [inline oommfc/scripts/evolver.py:83]:
```suggestion
```

**Comment by swapneelap** [comment]
> Is any of the tests actually running a simulation with time-varying `Xf_ThermSpinXferEvolver`? If not, we should add a test to `micromagnetictests`.

Yes absolutely! I will add tests. I am also thinking it would be good to reorder a few things in `oc.script` when it comes to simulations at finite temperature. For example, it is redundant to to ask user for an evolver if both `mm.Slonczewski` and `system.T` are defined since `oc.Xf_ThermSpinXferEvolver` is the only suitable option (?).

**Comment by swapneelap** [comment]
@lang-m if this is good for now, should we close this? There relevant test is [here](https://github.com/ubermag/micromagnetictests/blob/88a8bbae2c8831891b97d05ddee43d2b25162f20/micromagnetictests/calculatortests/slonczewski.py#L118-L129).

**Comment by lang-m** [comment]
@swapneelap Have you created an issue to remind us to rework the tests and get access to mif string? Once that is done, you can merge.

**Comment by swapneelap** [comment]
> @swapneelap Have you created an issue to remind us to rework the tests and get access to mif string? Once that is done, you can merge.

Done!

**Pull request merged**